### PR TITLE
fix: Default user profile is not attribute

### PR DESCRIPTION
**Pull request by MyvTsv**
## Checklist before requesting a review

*Please delete options that are not relevant.*

- [x] I have performed a self-review of my code.
- [x] I have added tests (when available) that prove my fix is effective or that my feature works.

## Description

Ticket #34792
The default user profile is not updated

### CODE REVIEW

File: inc/userinjection.class.php [hunk 1]
```diff
@@ -185,6 +185,45 @@ public function processAfterInsertOrUpdate($values, $add = true, $rights = [])
...
+                    in_array('users_id', $profile)
+                    && in_array('profiles_id', $profile)
+                    && in_array('entities_id', $profile)
+                ) {
+                    $is_new_profile_user = false;
+                    $profile_user->update([
+                        'id' => $profile['id'],
+                        'is_default_profile' => 1
+                    ]);
+                }
```
Comment (stonebuzz) [inline inc/userinjection.class.php:207]: This is always run (`$profile` alwys have `users_id`, `profiles_id`, `entities_id`)
so it's always updated `'is_default_profile' => 1`

File: inc/userinjection.class.php [hunk 2]
```diff
@@ -185,6 +185,45 @@ public function processAfterInsertOrUpdate($values, $add = true, $rights = [])
...
+                $profile_user->update([
+                    'id' => $profile['id'],
+                    'is_default_profile' => 0
+                ]);
```
Comment (stonebuzz) [inline inc/userinjection.class.php:211]: and this is always run too

**Review by stonebuzz — CHANGES_REQUESTED** [review]

**Review by Lainow — APPROVED** [review]

File: inc/userinjection.class.php [hunk 3]
```diff
@@ -185,6 +185,33 @@ public function processAfterInsertOrUpdate($values, $add = true, $rights = [])
             /** @phpstan-ignore-next-line */
             $DB->query($query); // phpcs:ignore
         }
+
+        if (isset($values['User']['profiles_id'])) {
+            $profile_user = new Profile_User();
+            $profile_user->getFromDBByCrit([
+                'users_id' => $values['User']['id'],
+                'entities_id' => $values['User']['entities_id'],
+                'prodiles_id' => $values['User']['profiles_id']
```
Comment (Rom1-B) [inline inc/userinjection.class.php:194]: I doubt it worked with the typo:
```suggestion
                'profiles_id' => $values['User']['profiles_id']
```
Can you retest the scenario described by @Lainow ?

**Review by Rom1-B — CHANGES_REQUESTED** [review]

**Comment by stonebuzz** [comment]
following user feedback i close

**Pull request closed without merging**